### PR TITLE
SCRUM-4647 implementation of secondary IDs for AGMs.

### PR DESCRIPTION
**Pull request by cmpich**

### CODE REVIEW

File: src/main/java/org/alliancegenome/curation_api/model/ingest/dto/AffectedGenomicModelDTO.java [hunk 1]
```diff
@@ -6,10 +6,13 @@
 import lombok.EqualsAndHashCode;
 import org.alliancegenome.curation_api.constants.LinkMLSchemaConstants;
 import org.alliancegenome.curation_api.interfaces.AGRCurationSchemaVersion;
+import org.alliancegenome.curation_api.model.ingest.dto.slotAnnotions.SecondaryIdSlotAnnotationDTO;
 import org.alliancegenome.curation_api.view.View;
 
 import java.util.List;
 
+import java.util.List;
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/model/ingest/dto/AffectedGenomicModelDTO.java:14]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.RedundantImportCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Duplicate import to line 12 - java.util.List.

File: src/main/java/org/alliancegenome/curation_api/model/entities/slotAnnotations/agmSlotAnnotations/AgmSecondaryIdSlotAnnotation.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import com.fasterxml.jackson.annotation.JsonBackReference;
+import jakarta.persistence.Entity;
+import jakarta.persistence.ManyToOne;
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import org.alliancegenome.curation_api.constants.LinkMLSchemaConstants;
+import org.alliancegenome.curation_api.interfaces.AGRCurationSchemaVersion;
+import org.alliancegenome.curation_api.model.entities.AffectedGenomicModel;
+import org.alliancegenome.curation_api.model.entities.Allele;
```
Comment (devopsagr) [inline src/main/java/org/alliancegenome/curation_api/model/entities/slotAnnotations/agmSlotAnnotations/AgmSecondaryIdSlotAnnotation.java:12]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused import - org.alliancegenome.curation_api.model.entities.Allele.

File: src/test/java/org/alliancegenome/curation_api/AgmBulkUploadITCase.java [hunk 3]
```diff
@@ -5,9 +5,12 @@
 import static org.hamcrest.Matchers.not;
 
 import java.time.OffsetDateTime;
+import java.util.ArrayList;
```
Comment (devopsagr) [inline src/test/java/org/alliancegenome/curation_api/AgmBulkUploadITCase.java:8]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused import - java.util.ArrayList.

File: src/test/java/org/alliancegenome/curation_api/AgmBulkUploadITCase.java [hunk 4]
```diff
@@ -5,9 +5,12 @@
...
 import java.util.List;
 
 import org.alliancegenome.curation_api.base.BaseITCase;
+import org.alliancegenome.curation_api.model.entities.AffectedGenomicModel;
```
Comment (devopsagr) [inline src/test/java/org/alliancegenome/curation_api/AgmBulkUploadITCase.java:12]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused import - org.alliancegenome.curation_api.model.entities.AffectedGenomicModel.

File: src/test/java/org/alliancegenome/curation_api/AgmBulkUploadITCase.java [hunk 5]
```diff
@@ -5,9 +5,12 @@
...
+import org.alliancegenome.curation_api.model.entities.slotAnnotations.agmSlotAnnotations.AgmSecondaryIdSlotAnnotation;
```
Comment (devopsagr) [inline src/test/java/org/alliancegenome/curation_api/AgmBulkUploadITCase.java:13]: 🚫 **[checkstyle]** <com.puppycrawl.tools.checkstyle.checks.imports.UnusedImportsCheck> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Unused import - org.alliancegenome.curation_api.model.entities.slotAnnotations.agmSlotAnnotations.AgmSecondaryIdSlotAnnotation.

File: src/main/resources/db/migration/v0.38.0.21__add_secondary_id_agm.sql [hunk 6]
```diff
@@ -0,0 +1,2 @@
+alter table slotannotation add column
+singleagm_id bigint;
```
Comment (oblodgett) [inline src/main/resources/db/migration/v0.38.0.21__add_secondary_id_agm.sql]: need a FK and index on this field

**Review by oblodgett — DISMISSED** [review]

**Review by markquintontulloch — APPROVED** [review]

File: src/main/resources/db/migration/v0.38.0.21__add_secondary_id_agm.sql (hunk 6 above)
Comment (cmpich) [reply src/main/resources/db/migration/v0.38.0.21__add_secondary_id_agm.sql]: Done

**Pull request merged**